Sources/SourceKitLSP/Swift/SwiftLanguageServer.swift

```diff
@@ -1299,6 +1299,30 @@ extension SwiftLanguageServer {
     return .full(RelatedFullDocumentDiagnosticReport(items: diagnostics))
   }
 
+  public func macroExpansion(_ req: MacroExpansionRequest) async throws -> MacroExpansion? {
+    let command = SemanticRefactorCommand(
+      title: "Expand Macro",
+      actionString: "source.refactoring.kind.expand.macro",
+      positionRange: req.range,
+      textDocument: req.textDocument
+    )
+    
+    do {
+      let refactor = try await semanticRefactoring(command)
+
+      guard let edit = refactor.edit.changes?[req.textDocument.uri]?.first else {
+        return nil
+      }
+
+      return MacroExpansion(
+        position: edit.range.lowerBound,
+        sourceText: edit.newText
+      )
+    } catch SemanticRefactoringError.noEditsNeeded(_) {
```

Comment:
Could this be simplified to 

```suggestion
    } catch SemanticRefactoringError.noEditsNeeded {
```